menas/ui/components/dataset/addDataset.fragment.xml

```diff
@@ -29,19 +29,43 @@
                         <core:Fragment type="XML" fragmentName="components.schema.selector.schemaSelector"/>
                         <Label text="HDFS raw data folder path"/>
                         <VBox>
-                        	<Input type="Text" id="selectedRawHDFSPathLabel"/>
+                            <HBox>
+                                <!-- HDFS Browser input -->
+                                <Input type="Text" id="selectedRawHDFSPathLabel" visible="{entity>/hdfsBrowserEnabled}">
+                                    <layoutData>
+                                        <FlexItemData growFactor="5"/>
```

Comment:
Interesting, what does this do? :)

I never learned how to use these 👍

Reply:
I wouldn't say I have learned how to use it, but here it allows input to grow beyond what it normally would. Without the `growFactor`, it would look like this:
![no-grow-factor](https://user-images.githubusercontent.com/4457378/107626249-cc489800-6c5d-11eb-9fdc-b0f66d3361a8.png)
(both elements in the `HBox` are only willing to grow to half of the width if allowed to grow at all)
Setting a higher `growFactor` has the effect of allowing the `Input` to grow beyond its fair-share space and thus, the `Input `takes all the free available space.

Reply:
Nice!